lambdas/bulk-migrate-vcs/src/main/java/uk/gov/di/ipv/core/bulkmigratevcs/BulkMigrateVcsHandler.java

```diff
@@ -139,6 +144,15 @@ public BatchReport handleRequest(Request request, Context context) {
                             USER_ID,
                             IDENTITY)) {
 
+                if ((report.getTotalMigrated() + page.count()) > EVCS_CLIENT_GOAWAY_LIMIT) {
```

Comment:
Will we ever hit this twice in one request? (i.e. when we hit 20k records)

I wonder if we should try going down the retry-in-EvcsClient approach 🤔

Reply:
Yeah, we actually might. And I've just noticed that as soon as we breach 10_000 this logic becomes silly.

Reply:
I've just pushed a change which should make it work properly. I think this'll be ok for now, without going down the retry route. I think adding retries to this script might actually make it a bit more unpredictable - if something fundamentally wrong happens we could end up backing ourselves off until the lambda times out.